src/services/authService.js

```diff
@@ -29,4 +31,47 @@ export default class AuthService {
 
     return { token, ...Helper.omitFields(user, ['password']) };
   }
+
+  /**
+ *
+ * @param {string} email email address
+ * @returns {string} token
+ */
+  static async forgotPassword(email) {
+    const result = await User.findOne({ where: { email } });
+    if (!result) throw new Error('Email not found');
+    const { dataValues: { first_name: name } } = result;
+    const token = await jwt.sign({
+      email
+    }, jwtSecret, { expiresIn: '1h' });
+    const emailDetails = {
+      receiver: email,
+      sender: 'chubi.best@gmail.com',
+      templateName: 'reset_password',
+      name,
+      confirm_account__url: `provide_newpass/reset_password/${token}`
+    };
+    await sendmail(emailDetails);
+    return token;
+  }
+
+  /**
+   *
+   * @param {string} token request token
+   * @param {string} password new password
+   * @returns {string} message
+   * */
+  static async resetPassword(token, password) {
+    const { email } = jwt.verify(token, jwtSecret);
```

Comment:
There is no check to ensure that the token belongs to the user.

Reply:
The verify method throws an error for an invalid token that is handled in the controller, the email was signed upon hitting the first route, so the verify method returns that email provided by the user in the first place... perhaps I need more context to understand what you want me to do